code/core/src/test/java/com/adobe/marketing/mobile/services/ui/AEPUIServiceTest.kt

```diff
@@ -96,4 +97,20 @@ class AEPUIServiceTest {
         assertEquals(alertEventListener, alertPresentable.getPresentation().eventListener)
         assertNotNull(alertPresentable.getPresentation().eventListener)
     }
+
+    @Test
+    fun `Test that #create throws an exception when application is null`() {
+        val iamSettings: InAppMessageSettings = InAppMessageSettings.Builder().build()
+        val iamEventListener: InAppMessageEventListener = mock(InAppMessageEventListener::class.java)
+        val iamPresentation = InAppMessage(iamSettings, iamEventListener)
+
+        `when`(mockPresentationUtilityProvider.getApplication()).thenReturn(null)
+
+        try {
```

Comment:
Can we use `assertThrows` instead?

Reply:
Ah. My bad. Missed this pointer. Will address as a followup/